crates/sui-core/src/consensus_adapter.rs

```diff
@@ -184,6 +186,78 @@ impl SubmitToConsensus for TransactionsClient<sui_network::tonic::transport::Cha
     }
 }
 
+/// A Narwhal client that instantiates LocalNarwhalClient lazily.
+pub struct LazyNarwhalClient {
+    /// Outer ArcSwapOption allows initialization after the first the connection to Narwhal.
+    /// Inner ArcSwap allows Narwhal restarts across epoch changes.
+    client: ArcSwapOption<ArcSwap<LocalNarwhalClient>>,
```

Comment:
IIUC ArcSwapAny cannot be copied and they need to be shared in an Arc<...> as well. So a Arc<ArcSwapAny<...>> needs to be shared between Narwhal and ConsensusAdapter. This client itself is also accessed concurrently, so another level of synchronization is needed, and I chose ArcSwapOption.

After the plumbing refactor, hopefully this will be removed.